src/wallet/wallet.cpp

```diff
@@ -1846,7 +1846,7 @@ bool CWalletTx::RelayWalletTransaction(CConnman* connman)
     return false;
 }
 
-std::set<uint256> CWalletTx::GetConflicts() const
+std::set<uint256> CWalletTx::GetConflicts() const EXCLUSIVE_LOCKS_REQUIRED(pwallet->cs_wallet)
```

Comment:
Add this to the header file instead?

Reply:
Moving it to the header file would generate a member access into an at that stage incomplete type `CWallet` I'm afraid. Any suggestion on what the best way of handling it would be?

Reply:
You can solve it by moving stuff around in the header file (i.e. make a forward declaration of CWalletTx, move the CWalletTx to the line after CWallet, move methods that still depend on the type to the cpp file, then add all the locking annotations. Note that just adding the annotation here is not sufficient, since I get a warning in `wallet/rpcwallet.cpp`...)

Personally, I had the feeling that c51b9e3 was a clear and obvious step in the right direction. I don't like to hold back the pull request based on that a single method was "missed". That can be fixed up later in a separate pull ... I'd suggest resetting hard to c51b9e3 and merge it

Reply:
@practicalswift @TheBlueMatt Any further thoughts here?

Reply:
@MarcoFalke Good point. I'll revert to c51b9e3.